configure.ac

```diff
@@ -762,23 +762,28 @@ Valid options are: yes, no, check, m32, mx32.])
 	[enable_mpers=yes])
 
 AS_IF([test x$arch = xaarch64],
-      [# So far, only aarch64 needs a separate compiler for its compat
-       # personality (which is AArch32, that is more or less ARMv7 EABI)
-       #
-       # Some heuristics regarding possible compiler name:
-       #   Debian:  arm-linux-gnueabi{,hf}-gcc
-       #   SuSE:    arm7{,hl}-linux-gnueabi-gcc
-       #   Red Hat: arm-redhat-linux-gnu-gcc
-       #   Fedora:  arm-linux-gnu-gcc
-       #   ALT:     armh-alt-linux-gnueabi-gcc
-       m4_foreach([triplet1], [arm, arm7, arm7hl, armh], dnl
-		  [m4_foreach([triplet2], [, $host_vendor-], dnl
-			      [m4_foreach([triplet3], [gnu, gnueabi, gnueabihf], dnl
-					  [m4_foreach([triplet_cc], [gcc, cc], dnl
-						      [m4_append([arm_compat_compilers], dnl
-								 triplet1[-]triplet2[linux-]triplet3[-]triplet_cc)])])])])
-       AC_CHECK_PROGS(CC_FOR_M32, arm_compat_compilers)
-       AS_IF([test -n "$CC_FOR_M32"],
+	[# So far, only aarch64 needs a separate compiler for its compat
+	# personality (which is AArch32, that is more or less ARMv7 EABI)
+	# First try $CC with 'aarch64' replaced with 'arm' and 'android'
+	# replaced with 'androideabi'
+	cc=${CC%/aarch64-*}/armv7a-${CC##*/aarch64-}
+	cc=${cc%-android*}-androideabi${cc##*-android}
```

Comment:
Why exactly should it take priority over all the other options? Also, since the `CC` differs from the one that ./configure would detect, I guess it is provided explicitly, why not provide `CC_FOR_M32` explicitly as well?